cms/server/contest/templates/user_test_row.html

```diff
@@ -76,13 +77,13 @@
         {% comment We replace '.%l' with the actual language only when it occurs as an extension at the end of the string and only when %}
         {% comment there isn't another file with that name. This allows to securily reverse the replacement and should work great in %}
         {% comment the common case. Yet, it still allows the marginal case of both 'foo.%l' and 'foo.c' in the submission format. %}
-        {% set files = t.files.keys() + t.managers.keys() %}
+        {% set files = list(itertools.chain(iterkeys(t.files), iterkeys(t.managers))) %}
```

Comment:
Why not list(a) + list(b) and one import less?

Reply:
Because I expected this to be more efficient in terms of memory and hence of running time. Yet I just did some tests and it turns out that they are essentially equal (actually, doing `list(chain)` is slower by a few percentage points). I'm reverting to `list + list`.